src/action.js

```diff
@@ -19,8 +19,18 @@ const Action = function (config) {
     throw new Error('Invalid arguments')
   }
   this.execute = (identity, model) => {
-    this.response = new Response(this.body(this, identity, model), this)
-    return Promise.resolve(this)
+    if (typeof this.body.then === 'function') {
+      return this.body(this, identity, model)
+        .then((data) => {
+          this.response = new Response(data, this)
+          return this
+        })
+    }
+    return Promise.all([])
```

Comment:
Thats a way of promisifying the rest of the body, instead I can do Promise.resolve, but it acts the same way